packages/hoppscotch-app/src/pages/profile.vue

```diff
@@ -344,6 +356,56 @@ const loadingCurrentUser = computed(() => {
   else return false
 })
 
+const showDeleteAccountModal = ref(false)
+const userVerificationInput = ref("")
+
+const adapter = new TeamListAdapter(true)
+const myAllTeams = useReadonlyStream(adapter.teamList$, null)
+
+const myTeams = computed(() => {
+  if (!myAllTeams.value) return []
+  return myAllTeams.value.filter((team) => {
+    return team.ownersCount === 1 && team.myRole === "OWNER"
+  })
+})
+
+const deletingUser = ref(false)
+
+const deleteUserAccount = async () => {
+  if (deletingUser.value) return
+  deletingUser.value = true
+
+  pipe(
+    deleteUser(),
+    TE.match(
+      (err: GQLError<string>) => {
+        deletingUser.value = false
+        toast.error(getErrorMessage(err))
+      },
+      () => {
+        deletingUser.value = false
+        showDeleteAccountModal.value = false
+        toast.success(t("settings.account_deleted"))
+        signOutUser()
+        router.push(`/`)
+      }
+    )
+  )()
+}
```

Comment:
The deletion work should absolutely move to a separate component.

Also, why are we fetching and calculating Teams list when the user is in this page itself ? This should be deferred to until the user actually opens the modal.

Hence, the finding owned teams and such should only happen when the modal is being summoned to prevent unnecessary work.

Reply:
The first UI approach was showing the team list on load. Not in the modal after clicking on Delete Account. That's why I had to calculate the team list on load. Still, it's my bad not to check after the UX changes by the reviewer.

I must separate the section for the user deletion feature into a component as I did with the `Shortcodes`.